### ISSUE TITLE
"metaget" support (mget, mset, mdelete)

### ISSUE THREAD

**Issue by dormando** [php-memcached-dev/php-memcached#442]
Yo!

https://github.com/memcached/memcached/pull/484 - I've been developing on this for a while, and after a break intend to finish it up. I'm personally happy with the general approach and the code is at the 90% mark. What I'd love to hear now is some feedback from client authors/maintainers on how good/bad/annoying this would be to implement.

I can clear up/rewrite text/add examples if it's not clear enough right now. The protocol is low level, and I'd expect a raw-ish interface from clients. Most users will only be needing a couple of high level patterns, which should be easy to implement wrapping the raw protocol (a herd-protected get, stale-while-revalidate, etc). I haven't written these examples up explicitly yet while I look for feedback.

Thanks!

**Comment by sodabrew** [comment]
A quick off-the-cuff: this looks awesome. But the structural problem is this client library is a wrapper for libmemcached, which hasn't been maintained in a while.

I don't have personal bandwidth or professional need to turn this client into its own protocol client, so unless libmemcached exposed APIs for the new commands I wouldn't add them.

**Comment by dormando** [comment]
That's correct, I'm trying to see if I can get libmemcached maintained again. Looks like there's a trickle of activity from Brian so it might not be too bad.

Either way, I still need feedback from people closer to implementing the API's in various languages :) If we were to get the API into libmemcached, etc?

**Comment by WPprodigy** [comment]
Latest on that: https://github.com/awesomized/libmemcached/issues/121
